### PR TITLE
feat: add email functionality to slugs update cmd and invalid slugs f…

### PR DESCRIPTION
**Pull request by AfaqShuaib09**
[PROD-3536](https://2u-internal.atlassian.net/browse/PROD-3536)
-----------------
This PR add slugs update summary report to `update_course_active_url_slugs_command` and add invalid slug format errors to report.

### CODE REVIEW

File: course_discovery/apps/course_metadata/management/commands/update_course_active_url_slugs.py [hunk 1]
```diff
@@ -56,26 +58,47 @@ def handle(self, *args, **options):
...
+                error_msg = f'Invalid course url slug: {course_url_slug}'
+                logger.error(error_msg)
+                self._add_records_to_update_slugs_summary_report(course_uuid, error_msg=error_msg)
                 continue
             try:
                 self.update_course_slug(course_uuid, course_url_slug)
             except Exception as exc:  # pylint: disable=broad-except
                 logger.error(f'Error occured while updating the course url slug.\n{exc}')
-
+        send_email_for_slug_updates(self._get_report_in_csv_format(), settings.NOTIFY_SLUG_UPDATE_RECIPIENTS)
```
Comment (DawoudSheraz) [inline course_discovery/apps/course_metadata/management/commands/update_course_active_url_slugs.py]: nit: override email subject to signify which mgt command has been run.

File: course_discovery/apps/course_metadata/management/commands/update_course_active_url_slugs.py [hunk 2]
```diff
@@ -56,26 +58,47 @@ def handle(self, *args, **options):
...
-                logger.error(f'Invalid course uuid: {course_uuid}')
+                error_msg = f'Invalid course uuid: {course_uuid}'
+                logger.error(error_msg)
+                self._add_records_to_update_slugs_summary_report(course_uuid=course_uuid, error_msg=error_msg)
                 continue
             if not course_url_slug or not is_valid_slug_format(course_url_slug):
-                logger.error(f'Invalid course url slug: {course_url_slug}')
+                error_msg = f'Invalid course url slug: {course_url_slug}'
+                logger.error(error_msg)
+                self._add_records_to_update_slugs_summary_report(course_uuid, error_msg=error_msg)
```
Comment (DawoudSheraz) [inline course_discovery/apps/course_metadata/management/commands/update_course_active_url_slugs.py:68]: update unit tests to accommodate this.

**Review by DawoudSheraz — APPROVED** [review]

File: course_discovery/apps/course_metadata/management/commands/update_course_active_url_slugs.py (hunk 1 above)
Comment (AfaqShuaib09) [reply course_discovery/apps/course_metadata/management/commands/update_course_active_url_slugs.py]: done

File: course_discovery/apps/course_metadata/management/commands/update_course_active_url_slugs.py (hunk 2 above)
Comment (AfaqShuaib09) [reply course_discovery/apps/course_metadata/management/commands/update_course_active_url_slugs.py:68]: fixed

File: course_discovery/apps/course_metadata/management/commands/update_course_active_url_slugs.py [hunk 3]
```diff
@@ -56,26 +58,53 @@ def handle(self, *args, **options):
...
+        send_email_for_slug_updates(
+            stats=self._get_report_in_csv_format(),
+            to_users=settings.NOTIFY_SLUG_UPDATE_RECIPIENTS,
+            subject='Course URL Slugs Update Report'
+        )
+
         self._log_report_in_csv_format()
 
-    def _log_report_in_csv_format(self):
+    def _get_report_in_csv_format(self):
```
Comment (Ali-D-Akbar) [inline course_discovery/apps/course_metadata/management/commands/update_course_active_url_slugs.py:83]: docstring

**Review by Ali-D-Akbar — APPROVED** [review]

**Pull request merged**